### PR TITLE
RPS Workaround

### PR DESCRIPTION
**Pull request by 333fred**
Workaround for RPS.

### CODE REVIEW

File: src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs [hunk 1]
```diff
@@ -98,14 +98,29 @@ public static async Task<RemoteHostClient> CreateAsync(HostWorkspaceServices ser
...
+                        (service, cancellationToken) => service.EnableAsync(AsynchronousOperationListenerProvider.IsEnabled, listenerProvider.DiagnosticTokensEnabled, cancellationToken),
+                        callbackTarget: null,
+                        cancellationToken).ConfigureAwait(false);
+                }
 
                 client.Started();
                 return client;
             }
+
+            static bool IsRpsMachine()
```
Comment (Cosifne) [inline src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs:113]: Looks like we are going to disable the OOP call and use the in proc client?

Comment (Cosifne) [reply src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs:113]: I wonder do we need to create a bug to track the progress of this.

Comment (333fred) [reply src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs:113]: Yes, that is correct.

**Review by Cosifne — APPROVED** [review]
I assume we have already tested in the VS side that this change works

**Comment by 333fred** [comment]
> I assume we have already tested in the VS side that this change works

Yes

File: src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/Remote/Core/ServiceHubRemoteHostClient.cs:113]: > Looks like we are going to disable the OOP call and use the in proc client?

No, we are disabling the ability of RPS to wait for Roslyn asynchronous operations to complete when those operations run in ServiceHub.

**Pull request merged**